devito/types/sparse.py

```diff
@@ -76,6 +82,67 @@ def __shape_setup__(cls, **kwargs):
             shape = (glb_npoint[grid.distributor.myrank],)
         return shape
 
+    def __distributor_setup__(self, **kwargs):
+        """
+        A `SparseDistributor` handles the SparseFunction decomposition based on
+        physical ownership, and allows to convert between global and local indices.
+        """
+        return SparseDistributor(kwargs['npoint'], self._sparse_dim,
+                                 kwargs['grid'].distributor)
+
+    def __subfunc_setup__(self, key, suffix):
+        if isinstance(key, SubFunction):
+            return key
+        elif key is not None and not isinstance(key, Iterable):
+            raise ValueError("`%s` must be either SubFunction "
+                             "or iterable (e.g., list, np.ndarray)" % key)
+
+        name = '%s_%s' % (self.name, suffix)
+        dimensions = (self.indices[self._sparse_position],
+                      Dimension(name='d'),
+                      Dimension(name='i'))
+        shape = (self.npoint, self.grid.dim, self._radius)
+
+        if key is None:
+            # Fallback to default behaviour
+            n = 2  # (Sparse points, Grid Dimensions)
+            dtype = self.dtype
+        else:
+            if not isinstance(key, np.ndarray):
+                key = np.array(key)
+                # Correct for corner case of single coordinate
+                n = max(key.ndim, 2)
+            else:
+                n = key.ndim
+            # Need to fix this check to get global npoint, global_shape broken
```

Comment:
Can we raise an error here if there's an unsupported use case? and perhaps add an x-failing MFE?

Reply:
It's not like there is an unsupported case it's just we need to add a way to get the global number of points, on the todo list

Reply:
I don't like commented out code, it's something we've never done.

It's perfectly fine to spell out it's a TODO... then maybe reference an issue on github and move the pseudocode in there?